packages/graphql/src/schema/validation/custom-rules/valid-types/node-missing-validation.ts

```diff
@@ -0,0 +1,180 @@
+import { asArray } from "../../../../utils/utils";
+import { assertValid, createGraphQLError, DocumentValidationError } from "../utils/document-validation-error";
+import { getPathToNode } from "../utils/path-parser";
+
+type ObjectExtensionsTypeMap = {
+    extensions: ObjectTypeExtensionNode[];
+    definition: ObjectTypeDefinitionNode;
+};
+/**
+ *  The below validation is here to help users to catch common issues
+ *  when using directives that are supposed to be used within the `@node` directive
+ *  but the `@node` directive is missing.
+ *  The below will raise an error if any of these cases are found:
+ *  if any of the following directive are found in a type that is not annotated with `@node`:
+ *  - authorization
+ *  - subscriptionAuthorization
+ *  - query
+ *  - mutation
+ *  - subscription
+ *  - fulltext
+ *  - vector
+ *  if any of the following directive are found in a field that is not annotated with `@node`:
+ *  - authorization
+ *  - subscriptionAuthorization
+ *  - relationship
+ *  - relayId
+ **/
+export function nodeMissingValidation(context: SDLValidationContext): ASTVisitor {
+    const extensionsTypeMap: Record<string, ObjectExtensionsTypeMap> = context
+        .getDocument()
+        .definitions.reduce((acc, def): Record<string, ObjectExtensionsTypeMap> => {
+            if (def.kind === Kind.OBJECT_TYPE_EXTENSION || def.kind === Kind.OBJECT_TYPE_DEFINITION) {
+                const typeName = def.name.value;
+                if (!acc[typeName]) {
+                    acc[typeName] = { extensions: [], definition: undefined };
+                }
+                if (def.kind === Kind.OBJECT_TYPE_EXTENSION) {
+                    if (acc[typeName].extensions) {
+                        acc[typeName].extensions.push(def);
+                    } else {
+                        acc[typeName].extensions = [def];
+                    }
+                } else {
+                    acc[typeName].definition = def;
+                }
+            }
+            return acc;
+        }, {});
+    return {
+        FieldDefinition(fieldDefinitionNode: FieldDefinitionNode, _key, _parent, path, ancestors) {
+            if (!fieldDefinitionNode.directives?.length) {
+                return;
+            }
+            const [_pathToNode, _traversedDef, parentOfTraversedDef] = getPathToNode(path, ancestors);
+            if (!parentOfTraversedDef) {
+                return;
+            }
+            const parentTypeAndExtensions = extensionsTypeMap[parentOfTraversedDef.name.value];
+            if (!parentTypeAndExtensions) {
+                return;
+            }
+            const allDirectives = [
+                ...(parentTypeAndExtensions.definition.directives ?? []),
+                ...parentTypeAndExtensions.extensions.flatMap((ext) => ext.directives ?? []),
+            ];
+            const nodeUsage = allDirectives?.find((directive) => directive.name.value === nodeDirective.name);
```

Comment:
Is this checking if the `@node` directive is being used? Maybe change the name to `hasNodeDirective` or `nodeDirective`?

Reply:
Yes, I'll change it!